chrome/content/zotero/xpcom/zotero.js

```diff
@@ -1156,12 +1151,41 @@ Components.utils.import("resource://gre/modules/osfile.jsm");
 			forceQuitNow ? null : Zotero.getString('general.restartLater'),
 			null, null, {});
 		
-		if (index == 0) {
+		if (forceQuitNow || index == 0) {
 			Services.startup.quit(Components.interfaces.nsIAppStartup.eAttemptQuit);
 		}
 		
 		return useProfileDir ? true : file;
 	}
+
+
+	this.warnOnUnsafeDataDir = true;
+	this.checkDataDirectory = function (path) {
```

Comment:
Let's actually call this `checkForUnsafeDataDirectory()` to be clear, since there are lots of other checks we could be doing.